webapp/components/posts_view.jsx

```diff
@@ -425,6 +425,15 @@ export default class PostsView extends React.Component {
 
         return false;
     }
+    componentWillUpdate(nextProps) {
+        if (this.props.postList && this.props.postList.order.length) {
```

Comment:
Good call, I think I once had a reason for it to be in WillUpdate, but I think I'd simplified since then. I also changed the location of one of the `if/else` conditions to cover an edge case of adding a user to a channel and the first message making the alert show.